src/ui/TemplateMessageItemBody/index.tsx

```diff
@@ -192,10 +190,14 @@ export function TemplateMessageItemBody({
             logger.error('TemplateMessageItemBody | no reservation key found in view_variables: ', reservationKey, templateData.view_variables);
             throw new Error();
           }
-          result.filledMessageTemplateItemsList = getFilledMessageTemplateItemsForCarouselTemplate(
-            simpleTemplateDataList,
-          );
-          result.carouselItem = carouselItem;
+          result.templateVersion = cachedTemplate.version;
+          result.filledMessageTemplateItemsList = [{
+            type: carouselItem.type as any,
```

Comment:
Do we have reason for converting its type to any here? I think it should be fine without this.

Reply:
No input is string but the type is a enum defined in the core package. The type is not exported so i could not import. I had to cast to any.